products/sentinel_one.py

```diff
@@ -725,60 +711,69 @@ def _process_queries(self) -> None:
         # execute queries in chunks
         # do not chunk if processing an IOC file
         ioc_hunt = list(self._queries.keys())
-        chunk_size = 1 if self._pq or (len(ioc_hunt) == 1 and ioc_hunt[0].tag.startswith('IOC - ')) else 10
+        chunk_size = 1 if (len(ioc_hunt) == 1 and ioc_hunt[0].tag.startswith('IOC - ')) else 10
 
-        with concurrent.futures.ThreadPoolExecutor(max_workers=25 if self._pq else 1) as executor:
+        with concurrent.futures.ThreadPoolExecutor(max_workers=25) as executor:
             futures = list[Future]()
 
-            # merge queries into one large query and execute it
-            for i in range(0, len(query_text), chunk_size):
-                # do not chain more than 10 ORs in a S1QL query
-                merged_tags = set[Tag]()
-                merged_query = ''
-                for tag, query_str in query_text[i:i + chunk_size]:
-                    # combine queries with ORs
-                    if merged_query:
-                        merged_query += ' OR '
-
-                    merged_query += query_str
-
-                    # add tags to set to de-duplicate
-                    merged_tags.add(tag)
-
-                # merge all query tags into a single string
-                merged_tag = Tag(','.join(tag.tag for tag in merged_tags),
-                                 ','.join(str(tag.data) for tag in merged_tags))
-
-                if self._query_base is not None and len(self._query_base):
-                    # add base_query filter to merged query string
-                    merged_query = f'{self._query_base} AND ({merged_query})'
-
-                if self._pq:
-                    # PQ seems to not honor site IDs provided in POST request body
-                    if len(self._site_ids):
-                        # restrict query to specified sites
-                        merged_query = f'({merged_query}) AND ('
-                        first = True
-                        for site_id in self._site_ids:
-                            if not first:
-                                merged_query += ' OR '
-                            else:
-                                first = False
+            tag_buckets = {}
+            # group built queries by tag
+            for item in query_text:
+                tag = item[0].tag
+                if tag in tag_buckets:
+                    tag_buckets[tag].append(item)
+                else:
+                    tag_buckets[tag] = [item]
```

Comment:
Since the variable `tag` has already been used, mypy throws errors. This rename should resolve that issue

```suggestion
                tag_value = item[0].tag
                if tag_value in tag_buckets:
                    tag_buckets[tag_value].append(item)
                else:
                    tag_buckets[tag_value] = [item]
```